core/src/cost_model.rs

```diff
@@ -0,0 +1,427 @@
+//! `cost_model` aiming to limit the size of broadcasting sets, and reducing the number
+//! of un-parallelizeble transactions (eg, transactions as same writable key sets).
+//! By doing so to improve leader performance.
+
+use crate::cost_tracker::CostTracker;
+use solana_sdk::{
+    bpf_loader, bpf_loader_deprecated, bpf_loader_upgradeable, clock::Slot, message::Message,
+    pubkey::Pubkey, system_program, transaction::Transaction,
+};
+use std::{collections::HashMap, str::FromStr};
+
+// TODO  revisit these hardcoded numbers, better get from mainnet log
+const COST_UNIT: u32 = 1;
+const DEFAULT_PROGRAM_COST: u32 = COST_UNIT * 500;
+const CHAIN_MAX_COST: u32 = COST_UNIT * 100_000;
+const BLOCK_MAX_COST: u32 = COST_UNIT * 100_000_000;
+
+use u32 as Cost;
+use Pubkey as ProgramKey;
```

Comment:
It looks a bit weird as `use` I think we would generally use `type Cost = u32;` for a type alias.